tracer/test/Datadog.Trace.Tests/Headers/NameValueHeadersCollectionTest.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="NameValueHeadersCollectionTest.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Specialized;
+using System.Linq;
+using Datadog.Trace.Headers;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Headers
+{
+    public class NameValueHeadersCollectionTest
```

Comment:
nit: Similar to the the previous `HeaderCollectionTests`, you could generate several `IHeadersCollection` objects and test that they all behave the same way instead of just the `NameValueHeadersCollection`